tests/integration/helm/install_test.go

```diff
@@ -0,0 +1,218 @@
+// with first-party-jwt enabled
+func TestInstallWithFirstPartyJwt(t *testing.T) {
+	framework.
+		NewTest(t).
+		Run(func(ctx framework.TestContext) {
+			workDir, err := ctx.CreateTmpDirectory("helm-install-test")
+			if err != nil {
+				t.Fatal("failed to create test directory")
+			}
+			cs := ctx.Environment().(*kube.Environment).KubeClusters[0]
+			h := helm.New(cs.Filename(), ChartPath)
+			s, err := image.SettingsFromCommandLine()
+			if err != nil {
+				t.Fatal(err)
+			}
+			overrideValuesStr := `
+global:
+  hub: %s
+  tag: %s
+  jwtPolicy: first-party-jwt
+`
+			overrideValues := fmt.Sprintf(overrideValuesStr, s.Hub, s.Tag)
+			overrideValuesFile := filepath.Join(workDir, "values.yaml")
+			if err := ioutil.WriteFile(overrideValuesFile, []byte(overrideValues), os.ModePerm); err != nil {
+				t.Fatalf("failed to write iop cr file: %v", err)
+			}
+			installIstio(t, ctx, cs, h, overrideValuesFile)
+
+			verifyInstallation(t, ctx, cs)
+
+			t.Cleanup(func() {
+				deleteIstio(t, cs, h)
+			})
+		})
+}
+
+// installIstio install Istio using Helm charts with the provided
+// override values file and fails the tests on any failures.
+func installIstio(t *testing.T, ctx resource.Context, cs resource.Cluster,
+	h *helm.Helm, overrideValuesFile string) {
+	if _, err := cs.CoreV1().Namespaces().Create(context.TODO(), &kubeApiCore.Namespace{
+		ObjectMeta: kubeApiMeta.ObjectMeta{
+			Name: IstioNamespace,
+		},
+	}, kubeApiMeta.CreateOptions{}); err != nil {
+		_, err := cs.CoreV1().Namespaces().Get(context.TODO(), IstioNamespace, kubeApiMeta.GetOptions{})
+		if err == nil {
+			log.Info("istio namespace already exist")
+		} else {
+			t.Errorf("failed to create istio namespace: %v", err)
+		}
+	}
+
+	// Install base chart
+	err := h.InstallChart(BaseReleaseName, BaseChart,
+		IstioNamespace, overrideValuesFile)
+	if err != nil {
+		t.Errorf("failed to install istio %s chart", BaseChart)
+	}
+
+	// Install discovery chart
+	err = h.InstallChart(IstiodReleaseName, filepath.Join(ControlChartsDir, DiscoveryChart),
+		IstioNamespace, overrideValuesFile)
+	if err != nil {
+		t.Errorf("failed to install istio %s chart", DiscoveryChart)
+	}
+
+	// Install ingress gateway chart
+	err = h.InstallChart(IngressReleaseName, filepath.Join(GatewayChartsDir, IngressGatewayChart),
+		IstioNamespace, overrideValuesFile)
+	if err != nil {
+		t.Errorf("failed to install istio %s chart", IngressGatewayChart)
+	}
+
+	// Install egress gateway chart
+	err = h.InstallChart(EgressReleaseName, filepath.Join(GatewayChartsDir, EgressGatewayChart),
+		IstioNamespace, overrideValuesFile)
+	if err != nil {
+		t.Errorf("failed to install istio %s chart", EgressGatewayChart)
+	}
+}
+
+// deleteIstio deletes installed Istio Helm charts and resources
+func deleteIstio(t *testing.T, cs resource.Cluster, h *helm.Helm) {
+	scopes.Framework.Infof("cleaning up resources")
+	if err := h.DeleteChart(IngressReleaseName, IstioNamespace); err != nil {
```

Comment:
you should do egress first as it was the last one installed, then ingress

Reply:
Good catch, resolved!